### PR TITLE
Fix scrollview momentum not stopping on scrollTo/scrollToEnd for horizontal scrollviews

### PR DESCRIPTION
**Pull request by Almouro**
### Motivation
My main motivation for this is using nested horizontal `Flashlists` inside a vertical `Flashlist`.
Like a `RecyclerView`, since my horizontal lists get recycled, if I scroll say the first horizontal list and scroll down, then when this list gets recycled it continues scrolling even if the content is new:

nested-list-recycling

To handle this, I want to call `scrollTo` everytime a new row appears to reset the scroll offset, however I've realized this doesn't stop the scroll momentum

### The bug

When scrolling and calling `scrollTo`, scroll momentum should be stopped and we should scroll to where `scrollTo` asked for.
All credit goes to @tomekzaw for https://github.com/facebook/react-native/pull/36104 who fixed it for vertical scrollviews

I realized we had the same issue for 
- horizontal scroll views 
- when calling `scrollToEnd`

| Vertical scrollview (working ✅) | Horizontal scrollview (before fix, not stopping ❌) | Horizontal scrollview (after fix ✅)|
|--------|--------|--|
| vertical-scrolltoffset-working|horizontal-scrolloffset-bug|horizontal-scrolloffset-fixed|

Based on  https://github.com/facebook/react-native/pull/38728 I kept all those calls to `abortAnimation` on the View Manager

## Changelog:

[ANDROID] [FIXED] - Fixed horizontal ScrollView momentum not stopping when calling scrollTo programmatically
[ANDROID] [FIXED] - Fixed ScrollView momentum not stopping when calling scrollToEnd programmatically

## Test Plan:

My test code is [this](https://gist.github.com/Almouro/3ffcfbda8e2239e64bee93985e243000)
Basically:
- a scrollview with a few elements
- some buttons to trigger a `scrollTo`

To reproduce the bug, I scroll then click one of the buttons triggering a `scrollTo`

I added `react-native@nightly` to my project, and copy pasted `packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll` folder from this branch to try out my fixes

Then tested the following scenarios on Android:

| List layout  | Method | Not animated | Animated|
|--------|--------|--|--|
| horizontal | scrollTo | horizontal-scrollTo-no| horizontal-scrollto-yes |
| horizontal | scrollToEnd  |horizontal-scrolltoend-no | horizontal-scrolltoend-yes|
| vertical | scrollTo | vertical-scrollto-no|vertical-scrollto-yes |
| vertical | scrollToEnd  | vertical-scrolltoend-no | vertical-scrolltoend-yes|

### CODE REVIEW

**Comment by lunaleaps** [comment]
This makes sense to me! Is there something we can add to RNTester to showcase the behavior?

**Comment by Almouro** [comment]
@lunaleaps I'm realizing there was already an example for this inside the RN Tester, I hadn't thought to check 😅 

| Before fix ❌ | After fix ✅ |
|--------|--------|
| rntester-beforefix| rntester-afterfix|
